manifests/compute/foreman_compute.pp

```diff
@@ -0,0 +1,13 @@
+# Provides a common class to install foreman-compute
+# used by multiple compute resources
+#
+# === Parameters:
+#
+# $version::  Package version to install, defaults to installed
+#
+class foreman::compute::foreman_compute ( $version = 'installed' ) {
```

Comment:
That limits the ability to pin a specific version of the foreman-compute module.  If it can be referenced in a manifest you can pin it for rackspace as an example 

````
class { '::foreman::compute::foreman_compute':
  version => '1.7.3',
}